### PR TITLE
contTables: Fisher's exact test can use a Monte Carlo simulation if the default work area is insufficient

### PR DESCRIPTION
**Pull request by MAgojam**
In contTables, if Fisher's exact test returns an error for an insufficient predefined work area, it is retried by activating with a Monte Carlo simulation to determine the p-value.

### CODE REVIEW

File: R/conttables.b.R [hunk 1]
```diff
@@ -492,6 +499,9 @@ contTablesClass <- R6::R6Class(
                 if (inherits(fish, 'htest') && all(dim(mat) == 2) && hypothesis != "different")
                     chiSq$addFootnote(rowNo=othRowNo, 'p[fisher]', hypothesisTested)
 
+                if (MCpsimul)
+                    chiSq$addFootnote(rowNo=othRowNo, 'p[fisher]', .('Monte Carlo simulation (B=2000)'))
```
Comment (jonathon-love) [inline R/conttables.b.R]: shift the ` (B=2000)` outside the translation string

File: R/conttables.b.R [hunk 2]
```diff
@@ -342,6 +342,14 @@ contTablesClass <- R6::R6Class(
                     dp <- NULL
                     lor <- NULL
                     fish <- try(stats::fisher.test(mat, conf.level=ciWidth, alternative=Ha), silent=TRUE)
+
+                    if (base::inherits(fish, 'try-error')) {
+                        # Monte Carlo simulation for p-value
+                        fish <- try(stats::fisher.test(mat, alternative=Ha, simulate.p.value=TRUE), silent=TRUE)
```
Comment (jonathon-love) [inline R/conttables.b.R:348]: do you need to specify the `conf.level` here? like the previous call?

File: R/conttables.h.R [hunk 3]
```diff
@@ -33,14 +33,16 @@ contTablesOptions <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
...
+            resS = FALSE,
             barplot = FALSE,
             yaxis = "ycounts",
             yaxisPc = "total_pc",
             xaxis = "xrows",
             bartype = "dodge", ...) {
 
             super$initialize(
-                package="jmv",
+                package="magojamtests",
```
Comment (jonathon-love) [inline R/conttables.h.R]: !!

File: R/conttables.h.R [hunk 4]
```diff
@@ -701,13 +729,13 @@ contTablesBase <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
                 requiresMissings = FALSE)
         }))
 
-#' Contingency Tables
+#' Tabella di Contingenza
```
Comment (jonathon-love) [inline R/conttables.h.R]: !!

File: R/conttables.h.R [hunk 5]
```diff
@@ -340,6 +357,17 @@ contTablesResults <- if (requireNamespace("jmvcore", quietly=TRUE)) R6::R6Class(
...
+                options=options,
+                name="phoc",
+                title="Post-hoc (residuals)",
+                visible="(resP || resS)",
+                columns=list(),
+                clearWith=list(
+                    "rows",
+                    "cols",
+                    "counts",
+                    "layers")))
```
Comment (jonathon-love) [inline R/conttables.h.R]: !!

**Review by jonathon-love — CHANGES_REQUESTED** [review]
hey mz,

sorry for the delay ... i think you've got some stray content in your .a.yaml file which is showing up in the .h.R file.

otherwise this is looking good.

**Comment by MAgojam** [comment]
> sorry for the delay ... i think you've got some stray content in your .a.yaml file which is showing up in the .h.R file.

Sorry, you're right, I hadn't noticed.
:sob: It won't happen again!
Unfortunately I passed (stupidly) the two conttables files .b.r and .h.r of a module that I use as a clone (so as not to use jmv) in which I am adding a Post-hoc for a significant chi-square with the Standardized and Adjusted residuals (for an upcoming listing proposal for a contingency table) .

File: R/conttables.b.R (hunk 1 above)
Comment (MAgojam) [reply R/conttables.b.R]: We can also eliminate `(B=2000)`, considering that it is the default value.

File: R/conttables.b.R (hunk 2 above)
Comment (MAgojam) [reply R/conttables.b.R:348]: I can put it back (as in the above) even if it is ignored in the Monte Carlo simulation.

File: R/conttables.h.R (hunk 4 above)
Comment (MAgojam) [reply R/conttables.h.R]: Last reply, below.

File: R/conttables.h.R (hunk 5 above)
Comment (MAgojam) [reply R/conttables.h.R]: Last reply, below.

File: R/conttables.b.R (hunk 1 above)
Comment (jonathon-love) [reply R/conttables.b.R]: yeah, lets eliminate for now.

File: R/conttables.b.R (hunk 2 above)
Comment (jonathon-love) [reply R/conttables.b.R:348]: oh ok, if it's not used, then i agree it isn't necessary

**Comment by jonathon-love** [comment]
merged. thanks mz!

**Pull request closed without merging**